quadlets/adguard/adguard.container

```diff
@@ -0,0 +1,28 @@
+[Unit]
+Description=Adguard Quadlet
+
+[Service]
+Restart=always
+TimeoutStartSec=900
+
+[Install]
+WantedBy=default.target
+
+[Container]
+Image=docker.io/adguard/adguardhome:latest
+ContainerName=adguard
+HostName=adguard
+
+# Optionally run this on network mode host, if you want real client ips to show in the logs
+# Network=host
+PublishPort=53:53/tcp
+PublishPort=53:53/udp
+PublishPort=784:784/udp
+PublishPort=853:853/tcp
+PublishPort=3000:3000/tcp
+PublishPort=8844:80/tcp
```

Comment:
Is this port mapping mismatched because your environment already uses port 80 for something else? Or is this an upstream example like from their compose file? I would like to avoid mapping ports around if possible to make it easier to drop in a single quadlet with no quirks.

Reply:
This is because these ports are in use, anyone running Nginx, OpenResty, Apache etc will have to do this.

Reply:
Okay this is a common issue when running multiple web servers. In my setup, I do not expose any ports except `80` and `443` with Caddy as a reverse proxy in front of all my containers. I want to try to make such a setup explicit and easy for people. My Caddyfile has a bunch of simple blocks like:

```
@my-app host my-app.mcgee.red
	handle @my-app {
		reverse_proxy my-app:80
	}
```

In fact, I do not use the `PublishPort` setting at all on mine, except exactly `80` and `443` for `caddy.container`. I don't see a better way to solve this problem of port map mazes. What do you think?

Reply:
I generally agree 100% in most use cases, however I feel adguard and pi-hole would be the exception to the rule here as most people run these locally within the network rather than exposing them and have them handle dns requests etc directly rather than having a middle man proxy involved, as if caddy goes down you would forfeit all dns.

Reply:
I probably still need to correct some of my other quadlets, but in this repo I'd like to show the upstream defaults as much as possible for the basic quadlet service. In advanced examples, I will add documentation regarding how to overcome port conflicts.

Does that make sense to you? I'm not asking you to add a commit, just curious if my idea sounds okay to someone who isn't myself.

Reply:
Additionally, the reverse proxy is only for the web ui frontens on ports `80` and/or `443`. The dns will still be handled without any proxying.

Reply:
Yeah that makes sense, if you want to keep it simple and have it run through a proxy, my only gripe would be that most people wont want to proxy a dns server and keep it local :)

Reply:
The proxy only applies to the frontend for the web ui. It won't affect the dns server or any other services.

Reply:
So you would say omit the 80 and 443 port publishing from the quadlet here and have the proxy handle only that portion? How would the proxy communicate with the adguard container on internal ports, in podman 5+ if you dont enable a `Network=` it will default to a pasta network driver so there will be no exposed ports for you communicate with.

Reply:
Publish 80:80 and 443:443 (and maybe `443:443/udp`). It's not feasible to remap those ports for every quadlet under the sun, right? If there are port conflicts, that is an issue that you have to resolve in your local environment. You can either remap them yourself, downstream in your local env, or you can proxy and/or load balance them. The main thing is that's not something an upstream example should define. 

However, as i said, I am working on advanced examples to demonstrate and document how to handle port conflicts among other common issues, such as re-using quadlets for different needs like running two `postgres` instances for different purposes.

Also, if you are remapping 80 and 443 often in your setup, I recommend putting caddy in front of it. You don't need a public domain. You can use it on your LAN just as effectively without exposing it to the web. Plus you can re-use the `Network=` key multiple times, so you can join 1 caddy instance to multiple networks. That caddy can then route you to all your web guis with a simple config, by container or host name. There are lots of ways to solve this problem, unless you just prefer to remap your host ports. You'll run out of ports eventually and have to start creating new virtual network interfaces to map more ports :P

Reply:
That would work by joining caddy to multiple networks without exposing ports, but most of the quadlet examples arent using defined networks so podman 5.0 defaults to pasta which means caddy has no networks to join to and therefore cant communicate with the containers ports internally through podman networks?